crowdsourcing/viewsets/project.py

```diff
@@ -67,68 +67,103 @@ def list(self, request, *args, **kwargs):
-                    ON up.id = wrr.target_id AND wrr.origin_type='worker' AND wrr.origin_id=%s
-                LEFT OUTER JOIN (SELECT target_id, AVG(CASE WHEN res.count=1 AND res.origin_id=%s
-                    THEN NULL ELSE res.weight END) AS average_requester_rating from
-                    (SELECT wr.*, count FROM crowdsourcing_workerrequesterrating wr
-                    INNER JOIN (SELECT target_id, COUNT(*) as count from crowdsourcing_workerrequesterrating
-                    WHERE origin_type='worker' GROUP BY target_id) temp ON wr.target_id=temp.target_id) res
-                    GROUP BY target_id) avg ON avg.target_id = up.id) calc WHERE owner_id<>%s
-                ) mod INNER JOIN crowdsourcing_project p ON p.id=mod.project_id
-                ORDER BY relevant_requester_rating desc;
+SELECT p.id, p.name, p.description, mod.id as module_id, mod.* FROM (
+
+SELECT id, name, description, created_timestamp, last_updated, owner_id, project_id, imputed_rating,
+    CASE WHEN real_weight IS NULL AND average_requester_rating IS NOT NULL THEN average_requester_rating
+    WHEN real_weight IS NULL AND average_requester_rating IS NULL THEN 1.99
+    WHEN real_weight IS NOT NULL AND average_requester_rating IS NULL THEN real_weight
+    ELSE real_weight + 0.1 * average_requester_rating END relevant_requester_rating
+    FROM (
+        SELECT rnk.*, wrr.weight as real_weight, avg.average_requester_rating FROM (
+
+--This fetches the modules according to cascading release
+SELECT evr.*
+FROM(
+    SELECT avgrat.*, CASE WHEN weight IS NULL
+        AND adj_average_worker_rating IS NOT NULL THEN adj_average_worker_rating
+        WHEN weight IS NULL AND adj_average_worker_rating IS NULL THEN 1.99
+        WHEN weight IS NOT NULL AND adj_average_worker_rating IS NULL THEN weight
+        ELSE weight + 0.1 * adj_average_worker_rating END worker_relevant_rating
+    FROM (
+        SELECT m.*, als.weight, als.adj_average_worker_rating, imputed_rating FROM crowdsourcing_module m
+            INNER JOIN crowdsourcing_requester r ON m.owner_id = r.id
+            INNER JOIN crowdsourcing_userprofile u ON r.profile_id = u.id
+            LEFT OUTER JOIN
+                (SELECT w.* FROM crowdsourcing_workerrequesterrating w
+                INNER JOIN(
+                    SELECT origin_id, MAX(last_updated) AS max_date FROM crowdsourcing_workerrequesterrating
+                        WHERE origin_type='requester' AND target_id = %s GROUP BY origin_id) tb
+                    ON w.origin_id = tb.origin_id AND w.last_updated = tb.max_date
+                    AND w.origin_type='requester' AND w.target_id=%s) w
+                ON u.id = w.origin_id
+            LEFT OUTER JOIN (
+                SELECT temp.origin_id, temp.target_id, temp.average_worker_rating, temp.count, temp.weight,
+                     (temp.average_worker_rating * temp.count - temp.weight) /
+                     (temp.count-1) as adj_average_worker_rating FROM
```

Comment:
gotcha,  cool
simplification suggestion no need to be considered now just in case you are interested
you can do a left outer join to wrrating on <your existing predicates> and origin_id<> owner_id, similarly as you did below, reduces number of rows processed and decreases the cost